### PR TITLE
add {generate,play} kube

### PR DESCRIPTION
**Pull request by vrothberg**
Add the `podman generate kube` and `podman play kube` command.  The code
has largely been copied from Podman v1 but restructured to not leak the
K8s core API into the (remote) client.

Both commands are added in the same commit to allow for enabling the
tests at the same time.

Move some exports from `cmd/podman/common` to the appropriate places in
the backend to avoid circular dependencies.

Move definitions of label annotations to `libpod/define` and set the
security-opt labels in the frontend to make kube tests pass.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>

### CODE REVIEW

File: cmd/podman/play/kube.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	if cmd.Flags().Changed("tls-verify") {
+		kubeOptions.SkipTLSVerify = types.NewOptionalBool(!kubeOptions.TLSVerifyCLI)
+	}
+	if kubeOptions.Authfile != "" {
+		if _, err := os.Stat(kubeOptions.Authfile); err != nil {
+			return errors.Wrapf(err, "error getting authfile %s", kubeOptions.Authfile)
+		}
+	}
+
+	_, err := registry.ContainerEngine().PlayKube(registry.GetContext(), args[0], kubeOptions.PlayKubeOptions)
```
Comment (vrothberg) [inline cmd/podman/play/kube.go:77]: We may just return an error.

File: cmd/podman/generate/kube.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	"github.com/containers/libpod/cmd/podman/utils"
+	"github.com/containers/libpod/pkg/domain/entities"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+var (
+	kubeOptions     = entities.GenerateKubeOptions{}
+	kubeFile        = ""
+	kubeDescription = `Command generates Kubernetes pod and service YAML (v1 specification) from a podman container or pod.
```
Comment (TomSweeneyRedHat) [inline cmd/podman/generate/kube.go:18]: 'podman container' -> "Podman container"

**Comment by rhatdan** [comment]
We can fix the issue presented by @TomSweeneyRedHat When you open podman remote PR.
/lgtm

**Comment by mheon** [comment]
/hold

Something's screwy with CI here

**Comment by vrothberg** [comment]
Added the remote endpoints and bindings. Local smoke tests were successful. I am currently adding bindings tests but wanted to throw it out for CI and reviews.

**Comment by vrothberg** [comment]
Something has changed in the tests deps. `vendor` was blocking before :(

**Comment by mheon** [comment]
Podman stats with json output flake
Restarting test

**Pull request merged**